### PR TITLE
feat(watson): add completion

### PR DESCRIPTION
**Pull request by tback**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

Add completion for watson time tracker https://tailordev.github.io/Watson/

## Other comments:
With #8847 there was a PR for watson. It was closed because of a missing Readme.

I'm a first time contributor. It'd be great if there were  links to the stuff I need to read in the PR template.

### CODE REVIEW

**Comment by tback** [comment]
This is getting a little stale. Please let me know if I can do anything to improve the situation.

File: plugins/watson/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Watson
+
+This plugin provides completion for [Watson](https://tailordev.github.io/Watson/).
+
+To use it add cake to the plugins array in your zshrc file.
```
Comment (carlosala) [inline plugins/watson/README.md:5]: It should be `watson` instead of `cake`.

File: plugins/watson/_watson [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#compdef watson
```
Comment (carlosala) [inline plugins/watson/_watson:1]: Could you check if there's any upgrade of it in the repo for this file?

**Review by carlosala — CHANGES_REQUESTED** [review]
You can delete the empty file `plugin.zsh`, it's not needed!

**Comment by carlosala** [comment]
@tback friendly ping :)

**Comment by carlosala** [comment]
@tback could you please fix those small things! Thanks! 😊

**Comment by carlosala** [comment]
For me the completion is not working, I'll need to figure out what's going on

**Comment by carlosala** [comment]
Thanks for your contribution!

**Comment by batbrain9392** [comment]
Is this an issue in `gh.plugin.zsh`?

`$ZSH_CACHE_DIR` is `~/.antigen/bundles/robbyrussell/oh-my-zsh/cache/`. Kindly note the trailing slash.
Then the completion path is used as `$ZSH_CACHE_DIR/completions/_gh`, which gives us `~/.../cache//completions/_gh`.
Would it be better if the path were `${ZSH_CACHE_DIR}completions/_gh`?

**Comment by batbrain9392** [comment]
But even after removing the extra slash, `gh` still doesn't work with antigen:

```
-- console output produced during zsh initialization follows --

~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/gh/gh.plugin.zsh:14: no such file or directory: ~/.antigen/bundles/robbyrussell/oh-my-zsh/cache//completions/_gh
```

Part of my `antigen` setup:
```
# Load the oh-my-zsh's library.
antigen use oh-my-zsh

# Bundles from the default repo (robbyrussell's oh-my-zsh).
antigen bundle git
antigen bundle macos
antigen bundle brew
antigen bundle gh  # trying to add this; rest works fine
```

**Comment by rwmitchell** [comment]
The double slash won’t cause an issue. Personally I’d prefer it removed from ZSH_CACHE_DIR and left explicit in the use, so it is obvious it is a subdir and not joining two strings

**Comment by batbrain9392** [comment]
> But even after removing the extra slash, `gh` still doesn't work with antigen:
> 
> ```
> -- console output produced during zsh initialization follows --
> 
> ~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/gh/gh.plugin.zsh:14: no such file or directory: ~/.antigen/bundles/robbyrussell/oh-my-zsh/cache//completions/_gh
> ```
> 
> Part of my `antigen` setup:
> 
> ```
> # Load the oh-my-zsh's library.
> antigen use oh-my-zsh
> 
> # Bundles from the default repo (robbyrussell's oh-my-zsh).
> antigen bundle git
> antigen bundle macos
> antigen bundle brew
> antigen bundle gh  # trying to add this; rest works fine
> ```

The issue goes away if I manually create the `completions` folder, because that was non-existent. I wonder if it could be autogenerated in `antigen`.

(Terribly sorry if this thread is not the one for this issue)

**Comment by rwmitchell** [comment]
Does the directory get completed if you use the OMZ installer and not antigen?

**Comment by batbrain9392** [comment]
Need to check. Altho, seems like my solution above didn't work ultimately. I'll have to check how to use `gh` with `antigen` in a different way than `antigen bundle gh`.

**Pull request merged**